spec/rubocop/cli/options_spec.rb

```diff
@@ -415,12 +420,39 @@ class SomeCop < Cop
               remaining_range = pending_cop_warning.length..-(inspected_output.length + 1)
               pending_cops = output[remaining_range]
 
-              expect(pending_cops).to include("Style/SomeCop: # new in 0.80\n  Enabled: true")
+              expect(pending_cops).to include("Style/SomeCop: # new in 1.80\n  Enabled: true")
 
               manual_url = output[remaining_range].split("\n").last
 
               expect(manual_url).to eq(versioning_manual_url)
             end
+
+            context "when the department's NewCops option is a version > VersionAdded" do
+              let(:department_option) { "NewCops: '1.81'" }
+
+              it 'does not display a pending cop warning for that cop' do
+                expect(output).to start_with(pending_cop_warning)
+                expect(output).not_to include("Style/SomeCop: # new in 1.80\n  Enabled: true")
+              end
+            end
+
+            context "when the department's NewCops option is a version < VersionAdded" do
+              let(:department_option) { "NewCops: '1.79'" }
+
+              it 'does display a pending cop warning for that cop' do
+                expect(output).to start_with(pending_cop_warning)
+                expect(output).to include("Style/SomeCop: # new in 1.80\n  Enabled: true")
+              end
+            end
+
+            context "when the department's NewCops option is a version == VersionAdded" do
+              let(:department_option) { "NewCops: '1.80'" }
+
+              it 'does not display a pending cop warning for that cop' do
+                expect(output).to start_with(pending_cop_warning)
+                expect(output).not_to include("Style/SomeCop: # new in 1.80\n  Enabled: true")
+              end
+            end
```

Comment:
Updating these tests for Rubocop 1.0+ highlighted how fragile they are:

* They rely on the actual Rubocop version and will need to be changed when it passes the versions hardcoded here
* They rely on very specific output formatting from the CLI
* If anything updates underneath them (or a mistake is made when updating the tests), the "expect not_to include" ones have a tendency to pass, even if they shouldn't be, or aren't testing anything useful

Reply:
You probably could stub out the version constant for the tests.

Reply:
I agree though that these tests are brittle, it’d be good if we could generate a data structure of cops that will be outputted and use that both to output and to test against, rather than testing the output itself. 

At the very least, instead of `expect not_to include` I would test *for* the specific full message so that it’ll fail if something changes, but this might get overly verbose.